lib/src/Uncertainty/Algorithm/WeightedExperiments/TensorProductExperiment.cxx

```diff
@@ -0,0 +1,187 @@
+  // Compute marginal nodes and weights
+  // Compute dimension
+  // Compute size for each marginal experiment.
+  LOGDEBUG(OSS() << "Compute marginal nodes and weights");
+  Collection<Sample> marginalNodesCollection(numberOfMarginalExperiments);
+  Collection<Point> marginalWeightsCollection(numberOfMarginalExperiments);
+  UnsignedInteger dimension = 0;
+  Indices marginalSizeIndices(numberOfMarginalExperiments);
+  for (UnsignedInteger i = 0; i < numberOfMarginalExperiments; ++i)
+  {
+    Point weights;
+    const Sample nodes(collection_[i].generateWithWeights(weights));
+    marginalNodesCollection[i] = nodes;
+    marginalWeightsCollection[i] = weights;
+    dimension += nodes.getDimension();
+    marginalSizeIndices[i] = nodes.getSize();
+  }
+  LOGDEBUG(OSS() << "dimension = " << dimension);
+  LOGDEBUG(OSS() << "marginalSizeIndices = " << marginalSizeIndices);
+  // Create index set
+  LOGDEBUG(OSS() << "Create index set");
+  Tuples tuples(marginalSizeIndices);
+  const IndicesCollection indicesCollection(tuples.generate());
+  // Compute nodes and weights
+  LOGDEBUG(OSS() << "Compute nodes and weights");
+  const UnsignedInteger size = indicesCollection.getSize();
+  LOGDEBUG(OSS() << "Total size = " << size);
+  weights_ = Point(size);
+  nodes_ = Sample(size, dimension);
+  for (UnsignedInteger i = 0; i < size; ++i)
+  {
+    weights_[i] = 1.0;
+    UnsignedInteger marginalExperimentIndex = 0;
+    for (UnsignedInteger j = 0; j < numberOfMarginalExperiments; ++j)
+    {
+      const UnsignedInteger marginalIndex = indicesCollection(i, j);
+      const Point marginalWeights(marginalWeightsCollection[j]);
+      weights_[i] *= marginalWeights[marginalIndex];
+      const Sample marginalNodes(marginalNodesCollection[j]);
+      UnsignedInteger marginalDimension = marginalNodes.getDimension();
+      for (UnsignedInteger k = 0; k < marginalDimension; ++k)
+      {
+        nodes_(i, marginalExperimentIndex) = marginalNodes(marginalIndex, k);
+        ++marginalExperimentIndex;
+      } // Loop for k over the dimensions
+    } // Loop for j over the marginal experiments
+  } // Loop for i over the n-D nodes
+}
+
+/* Method save() stores the object through the StorageManager */
+void TensorProductExperiment::save(Advocate & adv) const
+{
+  WeightedExperimentImplementation::save(adv);
+  adv.saveAttribute("collection_", collection_);
+}
+
+/* Method load() reloads the object from the StorageManager */
+void TensorProductExperiment::load(Advocate & adv)
+{
+  WeightedExperimentImplementation::load(adv);
+  adv.loadAttribute("collection_", collection_);
+}
+
+/* Distribution collection accessor */
+const TensorProductExperiment::WeightedExperimentCollection & TensorProductExperiment::getWeightedExperimentCollection() const
+{
+  return collection_;
+}
+
+/* Distribution collection accessor */
+void TensorProductExperiment::setWeightedExperimentCollection(const WeightedExperimentCollection & coll)
```

Comment:
```suggestion
void TensorProductExperiment::setWeightedExperimentCollection(const WeightedExperimentCollection & experiments)
```
In the API doc, you named this parameter "experiments", which creates an inconsistency in the HTML doc, where both names appear:

![image](https://user-images.githubusercontent.com/31989332/160253308-9d407f4e-a1d8-47b5-ba29-8831c563a6cd.png)

Perhaps the name also needs to change in the .cxx file, but I am not sure.

Reply:
This is true. I fix it for `TensorProductExperiment` and i create the issue #2000 for other classes.

Reply:
Fixed.